src/libYARP_OS/include/yarp/os/impl/BufferedConnectionWriter.h

```diff
@@ -526,6 +526,7 @@ class yarp::os::impl::ConnectionRecorder : public ConnectionReader,
 public:
     ConnectionRecorder() {
         reader = YARP_NULLPTR;
+        writer = YARP_NULLPTR;
```

Comment:
Let's leave this and the others like this for now, but in general it is better to change the constructor to use the initialization list... Let's see if coverity is happy with this fix, otherwise we might have to move it again